aspnet/mvc/models/validation.rst

```diff
@@ -1,8 +1,196 @@
+
+.. code-block:: html
+ :emphasize-lines: 5, 10
+
+  <form asp-action="Create">
+    <div class="form-horizontal">
+      <h4>Movie</h4>
+      <hr />
+      <div asp-validation-summary="ValidationSummary.ModelOnly" class="text-danger"></div>
+        <div class="form-group">
+          <label asp-for="ReleaseDate" class="col-md-2 control-label"></label>
+          <div class="col-md-10">
+            <input asp-for="ReleaseDate" class="form-control" />
+            <span asp-validation-for="ReleaseDate" class="text-danger" />
+          </div>		
+      </div>	
+    </div>  
+  </form>
+
+The tag helper above renders the HTML below. Notice that the ``data-`` attributes in the HTML output correspond to the data annotations for the ``ReleaseDate`` property. 
+
+.. code-block:: html
+
+  <form action="/movies/Create" method="post">
+    <div class="form-horizontal">
+      <h4>Movie</h4>
+      <hr />
+      <div class="text-danger"></div>	
+      <div class="form-group">
+        <label class="col-md-2 control-label" for="ReleaseDate">ReleaseDate</label>
+        <div class="col-md-10">
+          <input class="form-control" type="datetime" 
+          data-val="true" data-val-required="The ReleaseDate field is required." 
+          id="ReleaseDate" name="ReleaseDate" value="" />		
+          <span class="text-danger field-validation-valid" 
+          data-valmsg-for="ReleaseDate" data-valmsg-replace="true" />
+        </div>
+      </div>	  
+      </div>
+  </form>
+            
+Client-side validation prevents submission until the form is valid. The Submit button runs JavaScript that either submits the form or displays error messages. 
+
+IClientModelValidator
+---------------------
+You may create client side logic for your custom attribute, and JQuery's unobtrusive validation will execute it on the client for you automatically as part of validation. The first step is to control what values render in the ``data-`` attributes by implementing the ``IClientModelValidator`` interface as shown here:  
+
+.. literalinclude:: validation/sample/ClassicMovieAttribute.cs
+ :language: c#
+ :lines: 39-44
+ :dedent: 8
```

Comment:
@rachelappel Now that the dev branch is setup for RC2 we need to start thinking about getting this article updated with the RC2 changes. I'm ok with getting the current content into master for now, but please work with @dougbu on  the required RC2 updates in dev.